### PR TITLE
[Merged by Bors] - Modify Hare eligibility to use Tortoise beacon

### PR DESCRIPTION
**Pull request by lrettig**
## Motivation
Part of self-healing. Related to #2356. Removes another circular dependency in Hare eligibility: rather than relying on a layer pattern based on a set of contextually valid blocks from a safe layer, instead use the Tortoise beacon value.

## Changes
- Adds a simple, mock Tortoise beacon provider to node (waiting for #2267)
- Update Hare eligibility beacon to use this rather than a layer pattern for safe layer

## Test Plan
Existing tests updated. No new tests are required, as the Tortoise beacon is just mocked for now.

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: cmd/node/node.go [hunk 1]
```diff
@@ -590,7 +599,7 @@ func (app *SpacemeshApp) initServices(ctx context.Context,
 	if isFixedOracle { // fixed rolacle, take the provided rolacle
 		hOracle = rolacle
 	} else { // regular oracle, build and use it
-		beacon := eligibility.NewBeacon(mdb, app.Config.HareEligibility.ConfidenceParam, app.addLogger(HareBeaconLogger, lg))
+		beacon := eligibility.NewBeacon(tortoiseBeaconMock{}, app.Config.HareEligibility.ConfidenceParam, app.addLogger(HareBeaconLogger, lg))
```
Comment (noamnelke) [inline cmd/node/node.go:606]: I know it's relatively short-lived, but let's still add a `TODO` to replace this mock once the real beacon is implemented. I just can't stand to see the word "mock" in a production file with no explanation.

**Review by noamnelke — APPROVED** [review]

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge-

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merg

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Pull request closed without merging**